src/test/scala/uk/gov/nationalarchives/LambdaTest.scala

```diff
@@ -229,6 +232,45 @@ class LambdaTest extends ExternalServicesTestUtils with MockitoSugar {
       }
   }
 
+  forAll(identifierScenarios) {
+    (identifierFromDynamo, identifierFromPreservica, addIdentifierRequest, updateIdentifierRequest) =>
+      {
+        "handleRequest" should s"add ${addIdentifierRequest.map(i => s"${i.identifierName}=${i.value}").mkString(" ")} " +
+          s"and update ${updateIdentifierRequest.map(i => s"${i.identifierName}=${i.value}").mkString(" ")}" in {
+            val rowsWithIdentifiers = ListMap.from(folderIdsAndRows.map { case (id, dynamoResponse) =>
+              (id, dynamoResponse.copy(identifiers = identifierFromDynamo))
+            }.head :: Nil)
+
+            val mockLambda =
+              MockLambda(
+                convertFolderIdsAndRowsToListOfIoRows(rowsWithIdentifiers),
+                getIdentifiersForEntityReturnValues =
+                  IO(identifierFromPreservica.map(idp => IdentifierResponse("id", idp.identifierName, idp.value)))
+              )
+            mockLambda.handleRequest(mockInputStream, mockOutputStream, mockContext)
+
+            val updateCount = if (updateIdentifierRequest.isEmpty) 0 else 1
```

Comment:
1. Couldn't you do `updateIdentifierRequest.length` instead?
2. Does this need to be a variable?

Reply:
Not sure why I did this, You're right, I've got rid of the variable and used `updateIdentifierRequest.length`